### PR TITLE
feature: update single source endpoint

### LINKED ISSUE #168: FEAT - update single source endpoint
### What feature would you like to see?

I think it make more sense to update the single source endpoint from `/:sourceId` to `/sources/:sourceId` like the `regions` endpoint.

### Possible implementation

- change the endpoint for the single source id in `src/routes/api/news` 
- convert the `sourceId` comming from the request params to Lowercase and remove any spaces in `src/middlewares/existingSourceMiddleware

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by chinmaykumbhare**
## What type of PR is this? (check all applicable)

- [ :white_check_mark:] 💡 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/controller/api/newsAPIController.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ const controller = {
     return res.json(articles);
   },
   getNewsBySource: async (req, res) => {
-    const sourceId = req.params.sourceId;
+    const sourceId = (req.params.sourceId).toLowerCase();
```
Comment (MizouziE) [inline src/controller/api/newsAPIController.js:71]: Good thinking! :raised_hands:

File: src/middlewares/existingSourceMiddleware.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ const existingSources = sources.map((source) => source.name);
 
 //Finds sourceId match within existingSources, if not, return 404 page not found.
 function existingSourceMiddleware(req, res, next) {
-  const sourceId = req.params.sourceId;
+  const sourceId = (req.params.sourceId).toLowerCase();
```
Comment (MizouziE) [inline src/middlewares/existingSourceMiddleware.js:7]: :star2:

**Review by MizouziE — APPROVED** [review]
Well done catching the overlooked potential pitfall of case sensitivity!

**Review by MizouziE — CHANGES_REQUESTED** [review]
I have just realised that this gives cause for some documentation editing. 

in `API_DOCS.md` please change the following section:

image

You can change it as you see fit or use the following to replace the corresponding lines ~52:

```
### Articles by single source

[/api/news/sources/{source}](http://localhost:8000/api/news/sources/skynews)

These endpoints use the name of a single news source to retrieve **only the articles from that particular news source**. In the example above it will show for Sky News, but any of the avaiable sources can be used.

> **Note**
> The name of the source must have spaces omitted, otherwise it will return a 404 error.

```

**Comment by chinmaykumbhare** [comment]
have made the requested changes @MizouziE

**Review by MizouziE — APPROVED** [review]
Perfect. Thank you very much! 🙏

**Pull request merged**